client/src/components/search-input.tsx

```diff
@@ -1,19 +1,35 @@
 import { h } from "preact";
 
+export enum SearchInputType {
+  Default,
+  NumbersOnly,
+}
+
 export interface SearchInputProps {
   className: string,
   query: string,
   placeholder: string,
+  type: SearchInputType,
   onSearch: (query: string) => void,
 }
 
-export const SearchInput = ({ query, onSearch, className, placeholder }: SearchInputProps) => {
+export const SearchInput = ({ query, onSearch, className, placeholder, type }: SearchInputProps) => {
   const onInput = (event: Event) => {
     const target = event.target as HTMLInputElement;
     onSearch(target.value);
   };
 
-  return <input type='search' value={query} class={className}
+  const inputMode = mapToHtmlInputMode(type);
+  return <input type='search' inputMode={inputMode} value={query} class={className}
                 onInput={onInput}
                 placeholder={placeholder} autofocus={true} />
 }
+
+const mapToHtmlInputMode = (type: SearchInputType): string => {
+ switch (type) {
+   case SearchInputType.Default:
+     return 'search';
+   case SearchInputType.NumbersOnly:
+     return 'numeric';
+ }
+}
```

Comment:
Terminating newline is missing. Is there an error in our `.editorconfig`? Do we even have one? 🤔

Reply:
We don't have one 🤷‍♂

Reply:
d05dfd1